workflow/cron/controller.go

```diff
@@ -109,12 +109,15 @@ func (cc *Controller) runCronWorker() {
 }
 
 func (cc *Controller) processNextCronItem() bool {
+	cc.metrics.WorkerFree("cron-workflow")
 	key, quit := cc.cronWfQueue.Get()
 	if quit {
 		return false
 	}
 	defer cc.cronWfQueue.Done(key)
 
+	cc.metrics.WorkerBusy("cron-workflow")
```

Comment:
defer?

Reply:
can we name the queues the same?

Reply:
No need to defer, the worker is busy as soon as that line is hit. The worker becomes free again once the function is called again by the parent loop and the top line is hit (right before it block-waits)

Reply:
Yes, will work on naming

Reply:
I think it might be safer to do:

```golang
wfc.metrics.WorkerBusy("workflow_")
defer wfc.metrics.WorkerFree("workflow")
```

Reply:
That's what I had before, but my current solution has the added benefit of initializing the gauges to the total number of workers